compiler/src/dmd/parse.d

```diff
@@ -141,6 +141,18 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
 
         md = new AST.ModuleDeclaration(loc, a, id, msg, isdeprecated);
 
+        if (token.value == TOK.int32Literal)
+        {
+            auto edition = token.intvalue;
+            if (edition < Edition.min || Edition.max < edition)
+            {
+                error("module edition %lld must be in the range %d ... %d", edition, Edition.min, Edition.max);
+                edition = edition.min;
```

Comment:
This only works if editions are contigous which they may not be.

Something like this is going to handle it:

```d
            foreach(M; __traits(allMembers, Edition)[1 .. $-1]) {
                if (token.unsvalue == __traits(getMember, Edition, M))
                    goto EditionFound;
            }

            error(token.loc, "Unknown edition `%s`", token.toChars());
            goto EndOfEditionLex;

        EditionFound:
            mod.edition = cast(Edition)token.unsvalue;
```

Reply:
I expect we can course correct as our use of editions evolves. Right now, I'm trying to make it as simple as possible.

Your example looks good, except that it doesn't list what editions are supported.

Reply:
My concern is that we'll forget about it and it'll end up being a very nasty surprise for someone at some point!